Improved efficiency of Gsl2Fitter

tking53 opened the pull request.

spaulaus committed:
Improved efficiency of Gsl2Fitter

ksmith0 reviewed (dismissed):
Is there a reason you removed the virtual from the GslFitter methods?

There is no need to add `globals_` at the class scope. It is only used in `FittingAnalyzer::Analyze` and should be scoped there. This reduces the dependencies of the header.

ksmith0 commented on Scan/utkscan/analyzers/source/Gsl2Fitter.cpp:
```diff
@@ -89,21 +83,20 @@ void GslFitter::PerformFit(const std::vector<double> &data,
 
     gsl_multifit_fdfsolver_wset (s, &f, &x.vector, &w.vector);
-    gsl_multifit_fdfsolver_driver(s, 1000, xtol, gtol, ftol, &info);
+    gsl_multifit_fdfsolver_driver(s, 100, xtol, gtol, ftol, &info);
```
What does this hard-coded number do? Why did it change from 1000 to 100?

spaulaus replied:
That number is the maximum number of iterations, the major change is reducing the tolerance from 10^-10 to 10^-4.

spaulaus commented:
That number is the maximum number of iterations, the major change is reducing the tolerance from 10^-10 to 10^-4.

spaulaus committed:
Named a magic number in the Gsl2Fitter.cpp

I have addressed a magic number in the Gsl2Fitter, the magic number 100 was actually the maximum number of iterations.

spaulaus committed:
Addressing scoping change in Fitting Analyzer.

The globals_ variable had been added for simplicity, but was only used in a single instance so created unnecessary code bloat.

tking53 commented:
i get this error when i try to compile on Quack (gsl 1.13) 

```
git-testing/paass/Scan/utkscan/analyzers/source/Gsl1Fitter.cpp: In member function ‘virtual void GslFitter::PerformFit(const std::vector<double, std::allocator<double> >&, const std::pair<double, double>&, const bool&, const double&, const double&)’:
git-testing/paass/Scan/utkscan/analyzers/source/Gsl1Fitter.cpp:108: error: ‘isFastSipm_’ was not declared in this scope
```

spaulaus committed:
Removed unnecessary header and fixed if statement.

tking53 committed:
Fixes compilation issues with Gsl1Fitter

spaulaus commented:
@ksmith0 The requested changes (and more) have been made and tested.

spaulaus reviewed (approved):
The issues originally raised have been addressed. The magic number is now hard coded, but no longer magic.
